### PR TITLE
Upgrade with cms related changes and tests

### PR DESCRIPTION
**Pull request by kookster**
In the process of refactoring cms.prx.org to use this gem:
* copy fixes added to the similar code in cms to this gem 
* move and refactor tests to this gem

### CODE REVIEW

File: test/hal_api/concerns/link_serialize_test.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# encoding: utf-8
+require 'test_helper'
+require 'hal_api/representer'
+require 'hal_api/representer/link_serialize'
+
+describe HalApi::Representer::LinkSerialize do
+
+  it 'sets a default curie' do
```
Comment (cavis) [inline test/hal_api/concerns/link_serialize_test.rb]: is that really what you're testing here?

File: hal_api-rails.gemspec [hunk 2]
```diff
@@ -30,4 +30,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "bundler", "~> 1.10"
   spec.add_development_dependency "rake", "~> 10.0"
   spec.add_development_dependency "minitest"
+  spec.add_development_dependency "kaminari"
+  spec.add_development_dependency "pundit"
```
Comment (cavis) [inline hal_api-rails.gemspec]: Is the pundit/current_user/authorization stuff I see in here necessary?  (Asking because some of it is just commented out).

**Comment by cavis** [comment]
Looks pretty good to me.  And vaguely familiar!

I know this is mostly a refactor, but 2 issues to re-verify in CMS after getting this merged:

- PRX/cms.prx.org#136 "Zoom param not working"
- PRX/cms.prx.org#135 "Tags over-cached on update requests"

File: hal_api-rails.gemspec (hunk 2 above)
Comment (kookster) [inline hal_api-rails.gemspec]: The kaminari stuff is definitely used. I have commented out the ActiveRecord related code.

Comment (kookster) [inline hal_api-rails.gemspec]: Looking at it again, I can probably get the code to work w/o pundit.

**Comment by kookster** [comment]
@cavis better?

**Comment by cavis** [comment]
Yesssss.

**Pull request merged**